lib/ide/src/controller/module.rs

```diff
@@ -51,6 +63,12 @@ pub struct Metadata {
 
 impl parser::api::Metadata for Metadata {}
 
+/// Metadata used within ide.
+#[derive(Debug,Clone,Default,Serialize,Deserialize)]
+pub struct IdeMetadata {
+    /// Position in x,y coordinates for each Id.
+    position: HashMap<ID,Position>
```

Comment:
I'd create `NodeMetadata` with `Position`.
While this will be only node metadata right now, module controller should ont get new methods each time we want to add a new field.
Or perhaps `IdeMetadata` should store it as `<ID,serde_json::Value>` and have graph controller (or even graph view) deal with the stored value.